code/src/sixsq/nuvla/server/resources/module.clj

```diff
@@ -215,136 +215,101 @@ component, or application.
+              create-content
+              (dissoc :content)
+              (utils/set-price existing-module (auth/current-active-claim request)))))))
+
 (defmethod crud/edit resource-type
-  [{:keys [body] :as request}]
+  [request]
   (try
-    (let [id          (str resource-type "/" (-> request :params :uuid))
-          [module-meta
-           {:keys [author commit] :as module-content}] (-> body
-                                                           u/strip-service-attrs
-                                                           utils/split-resource)
-          {:keys [subtype versions price acl]} (crud/retrieve-by-id-as-admin id)
-          module-meta (-> module-meta
-                          (dissoc :parent-path :published)
-                          (assoc :subtype subtype)
-                          utils/set-parent-path)]
-
-      (a/can-edit? {:acl acl} request)
-
-      (if (utils/is-project? subtype)
-        (->> module-meta
-             (assoc request :body)
-             edit-impl)
-        (let [_              (throw-cannot-access-registries-or-creds request)
-              content-url    (subtype->resource-url subtype)
-
-              content-body   (some-> module-content (merge {:resource-type content-url}))
-
-              content-id     (when content-body
-                               (-> {:params      {:resource-name content-url}
-                                    :body        content-body
-                                    :nuvla/authn auth/internal-identity}
-                                   crud/add
-                                   :body
-                                   :resource-id))
-
-              versions       (when content-id
-                               (conj versions
-                                     (cond-> {:href   content-id
-                                              :author author}
-                                             commit (assoc :commit commit))))
-              price-changed? (and config-nuvla/*stripe-api-key*
-                                  (or (not= (:cent-amount-daily price)
-                                            (get-in module-meta [:price :cent-amount-daily]))
-                                      (not= (:currency price)
-                                            (get-in module-meta [:price :currency]))))]
-          (-> request
-              (update-in [:cimi-params :select] disj "compatibility")
-              (assoc :body
-                     (cond-> module-meta
-                             price-changed? (-> (assoc :price (merge price (:price module-meta)))
-                                                (utils/set-price (auth/current-active-claim request)))
-                             versions (assoc :versions versions)))
-              edit-impl))))
+    (-> request
+        throw-cannot-access-registries-or-creds
+        throw-project-cannot-have-price
+        throw-project-cannot-have-content
```

Comment:
Why no check for compatibilty via `throw-compatibility-required-for-application`? Would it mess up legacy apps?

Reply:
Yes. Compatibility is now set by user. For old ones, that doesn't have this flag we keep the old behavior until the user decide to edit the compatibility.